config/file_test.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func Test_LookupAuthConfig_WithNoConfigFile(t *testing.T) {
+	DefaultDir, _ = ioutil.TempDir("", "faas-cli-file-test")
+	DefaultFile = "test1.yml"
+	_, _, err := LookupAuthConfig("http://openfaas.test1")
+	if err == nil {
+		t.Errorf("Error was not returned")
+	}
+
+	r := regexp.MustCompile(`(?m:config file not found)`)
+	if !r.MatchString(err.Error()) {
+		t.Errorf("Error not matched: %s", err.Error())
+	}
+}
+
+func Test_UpdateAuthConfig_Insert(t *testing.T) {
+	DefaultDir, _ = ioutil.TempDir("", "faas-cli-file-test")
+	DefaultFile = "test2.yml"
+	u := "admin"
+	p := "some pass"
+	gatewayURL := strings.TrimRight("http://openfaas.test/", "/")
+	UpdateAuthConfig(gatewayURL, u, p)
+
+	user, pass, err := LookupAuthConfig(gatewayURL)
+	if err != nil {
+		t.Errorf("got error %s", err.Error())
+	}
+
+	if user != u || pass != p {
+		t.Errorf("got user %s and pass %s, expected %s %s", user, pass, u, p)
+	}
+}
+
+func Test_UpdateAuthConfig_Update(t *testing.T) {
```

Comment:
Thank you for taking the time to add the test coverage. This means a lot @stefanprodan